File: app/src/main/java/mchehab/com/androidlinearregression/TFSimpleRegression.java
package mchehab.com.androidlinearregression;

import android.content.Context;

import org.tensorflow.contrib.android.TensorFlowInferenceInterface;


public class TFSimpleRegression {
    static {
        System.loadLibrary("tensorflow_inference");
    }

    private TensorFlowInferenceInterface inferenceInterface;
    private static final String MODEL_FILE = "file:///android_asset/frozen.pb";
    private static final String INPUT_NODE = "input";
    private static final String[] OUTPUT_NODES = {"prediction"};
    private static final String OUTPUT_NODE = "prediction";
    private static final long[] INPUT_SIZE = {1};
    private static final int OUTPUT_SIZE = 1;

    TFSimpleRegression(final Context context) {
        inferenceInterface = new TensorFlowInferenceInterface(context.getAssets(), MODEL_FILE);
    }

    public float[] predict(float[] data) {
        float[] result = new float[OUTPUT_SIZE];
        inferenceInterface.feed(INPUT_NODE, data, INPUT_SIZE);
        inferenceInterface.run(OUTPUT_NODES);
        inferenceInterface.fetch(OUTPUT_NODE, result);

        return result;
    }
}

File: kotlin/src/main/java/mchehab/com/kotlin/TFSimpleRegression.kt
package mchehab.com.kotlin

import android.content.Context
import org.tensorflow.contrib.android.TensorFlowInferenceInterface

class TFSimpleRegression internal constructor(context: Context) {

    companion object {
        init {
            System.loadLibrary("tensorflow_inference")
        }
    }

    private val MODEL_FILE = "file:///android_asset/frozen.pb"
    private val INPUT_NODE = "input"
    private val OUTPUT_NODES = arrayOf("prediction")
    private val OUTPUT_NODE = "prediction"
    private val INPUT_SIZE = longArrayOf(1)
    private val OUTPUT_SIZE = 1

    private val inferenceInterface: TensorFlowInferenceInterface

    init {
        inferenceInterface = TensorFlowInferenceInterface(context.assets, MODEL_FILE)
    }

    fun predict(data: FloatArray): FloatArray {
        val result = FloatArray(OUTPUT_SIZE)
        inferenceInterface.feed(INPUT_NODE, data, *INPUT_SIZE)
        inferenceInterface.run(OUTPUT_NODES)
        inferenceInterface.fetch(OUTPUT_NODE, result)

        return result
    }
}